tests/components/rest/test_binary_sensor.py

```diff
@@ -88,6 +88,42 @@ def test_setup_minimum(self, mock_req):
             self.hass.block_till_done()
         assert 1 == mock_req.call_count
 
+    @requests_mock.Mocker()
+    def test_setup_minimum_resource_template(self, mock_req):
+        """Test setup with minimum configuration (resource_template)."""
+        mock_req.get("http://localhost", status_code=200)
+        with assert_setup_component(1, "binary_sensor"):
+            assert setup_component(
+                self.hass,
+                "binary_sensor",
+                {
+                    "binary_sensor": {
+                        "platform": "rest",
+                        "resource_template": "http://localhost",
+                    }
+                },
+            )
+            self.hass.block_till_done()
+        assert mock_req.call_count == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5745133)